src/core/service.c

```diff
@@ -2695,6 +2698,13 @@ static void service_sigchld_event(Unit *u, pid_t pid, int code, int status) {
                                 /* Fall through */
 
                         case SERVICE_RUNNING:
+                                if (s->type == SERVICE_NOTIFY) {
```

Comment:
Oh, now I remember! We want to fail when the main process exits unsuccessfully. Would you say that this should be part of `service_enter_running`? I'll draft something.